src/functions.php

```diff
@@ -1,33 +1,118 @@
 <?php
 
-function getAllGames(object $db): array
+function getAllGames(PDO $db): array
 {
     $query = $db->prepare(
-        'SELECT `name`, `genre`.`genre`, `platform`.`platform`, `image`, `agerating`.`age` FROM videogames
+        'SELECT `name`, `genre`.`genre`, `platform`.`platform`, `image`, `agerating`.`agerating` FROM videogames
         JOIN `genre` ON `videogames`.`genreid` = `genre`.`id`
         JOIN `platform` ON `videogames`.`platformid` = `platform`.`id`
         JOIN `agerating` ON `videogames`.`ageid` = `agerating`.`id`;'
     );
     $result = $query->execute();
-    if ($result) {
-        $videogames = $query->fetchAll();
-    } else {
+    if (!$result) {
         throw new ErrorException("Couldn't retrieve information from database");
+    } else {
+        $videogames = $query->fetchAll();
     }
     return $videogames;
 }
 
+function checkImgExists(array $game): string {
+    if (!$game['image']) {
+        return "";
+    } else {
+        return "<img class='card-img' src='{$game['image']}' alt='Image of {$game['name']} cover'>";
+    }
+}
 function displayAllGames(array $games): string
 {
     $result = "";
     foreach ($games as $game) {
+        $img = checkImgExists($game);
         $result .= "<div class='card'>
-        <img class='card-img' src='{$game['image']}' alt='Image of {$game['name']} cover'>
+        {$img}
         <h2>Game: {$game['name']}</h2>
         <p><strong>Genre:</strong> {$game['genre']}</p>
         <p><strong>Platform:</strong> {$game['platform']}</p>
-        <p><strong>Age Rating:</strong> {$game['age']}</p>
+        <p><strong>Age Rating:</strong> {$game['agerating']}</p>
     </div>";
     }
     return $result;
+}
+
+function getTable(string $tablename, PDO $db): array {
+    $query = $db->prepare("SELECT `id`, `{$tablename}` FROM `{$tablename}`");
```

Comment:
You should never concatenate strings into your SQL queries even if you are providing the concatenated stuff internally - you might be careful about using only table names you know, but another developer on your team might not be so careful and could expose this query to user input.

Unfortunately the only solution is to use different functions for each of these dropdown queries - you can't use placeholders for table names either.

Reply:
Done